### PR TITLE
test: e2e metrics test and refactor

### LINKED ISSUE #23979: [e2e] Signature redesign - Create metric tests
ref: https://github.com/MetaMask/metamask-extension/issues/23552#issuecomment-2017496179

Create e2e tests:
- [ ] Metrics
  - [ ] check we are tracking the redesigned UI is being used (while we have both new and old co-existing). For this we can add the value redesigned_confirmation to the ui_customizations property of all signature events when the redesigned UI is being displayed
  - [ ] check we are tracking opening of account info modal

### PR DESCRIPTION
**Pull request by pnarayanaswamy**
## **Description**

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/25632?quickstart=1)

## **Related issues**

Fixes: [#23979](https://github.com/MetaMask/metamask-extension/issues/23979)

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: test/e2e/tests/confirmations/signatures/sign-typed-data-v3.spec.ts [hunk 1]
```diff
@@ -102,14 +146,16 @@ async function assertInfoValues(driver: Driver) {
...
+  await driver.switchToWindowWithTitle('E2E Test Dapp', windowHandles);
   await driver.clickElement('#signTypedDataV3Verify');
+  await driver.delay(500);
 
   const verifyResult = await driver.findElement('#signTypedDataV3Result');
-  await driver.waitForSelector({
-    css: '#signTypedDataV3VerifyResult',
-    text: publicAddress,
-  });
+  // await driver.waitForSelector({
```
Comment (pedronfigueiredo) [inline test/e2e/tests/confirmations/signatures/sign-typed-data-v3.spec.ts:155]: remove?

File: test/e2e/tests/confirmations/signatures/permit.spec.ts [hunk 2]
```diff
@@ -56,6 +94,7 @@ describe('Confirmation Signature - Permit', function (this: Suite) {
         await driver.clickElement(
           '[data-testid="confirm-footer-cancel-button"]',
         );
+        await driver.delay(1000);
```
Comment (digiwand) [inline test/e2e/tests/confirmations/signatures/permit.spec.ts:97]: Q: this was added after observed flakiness?

File: test/e2e/tests/confirmations/helpers.ts [hunk 3]
```diff
@@ -31,7 +36,53 @@ export function withRedesignConfirmationFixtures(
...
+      .withJsonBodyIncluding({
+        batch: [{ type: 'track', event: 'Account Details Opened' }],
+      })
+      .thenCallback(() => {
+        return {
+          statusCode: 200,
+        };
+      }),
+  ];
+}
```
Comment (jpuri) [inline test/e2e/tests/confirmations/helpers.ts:88]: NIT: a loop can be created over event here.

File: test/e2e/tests/confirmations/signatures/sign-typed-data-v3.spec.ts [hunk 4]
```diff
@@ -102,14 +146,16 @@ async function assertInfoValues(driver: Driver) {
...
+  //   css: '#signTypedDataV3VerifyResult',
+  //   text: publicAddress,
+  // });
```
[the remark this answers is not in the document]
Comment (pnarayanaswamy) [reply test/e2e/tests/confirmations/signatures/sign-typed-data-v3.spec.ts:158]: ah I forgot to remove the commented code, will update it

File: test/e2e/tests/confirmations/signatures/sign-typed-data-v3.spec.ts (hunk 1 above)
Comment (pnarayanaswamy) [reply test/e2e/tests/confirmations/signatures/sign-typed-data-v3.spec.ts:155]: ah I forgot to remove the commented code, will update, thanks

**Review by jpuri — DISMISSED** [review]

File: test/e2e/tests/confirmations/signatures/sign-typed-data-v3.spec.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (pnarayanaswamy) [reply test/e2e/tests/confirmations/signatures/sign-typed-data-v3.spec.ts]: done

File: test/e2e/tests/confirmations/signatures/sign-typed-data-v3.spec.ts (hunk 1 above)
Comment (pnarayanaswamy) [reply test/e2e/tests/confirmations/signatures/sign-typed-data-v3.spec.ts]: done

**Review by pedronfigueiredo — APPROVED** [review]

**Review by OGPoyraz — APPROVED** [review]

**Pull request merged**